src/main/java/MisterDuke.java

```diff
@@ -0,0 +1,687 @@
+import java.io.*;
```

Comment:
I've noticed that this file is very long. You may want to consider putting different classes in separate files.